### PR TITLE
Add pyright hook to CI

### PR DESCRIPTION
**Pull request by dbrattli**

### CODE REVIEW

File: pyrightconfig.json [hunk 1]
```diff
@@ -1,3 +1,8 @@
 {
-  "exclude": ["expression/_version.py"]
+  "exclude": [
+    "expression/_version.py",
+    ".eggs",
+    "node_modules",
+    "oryx"
```
Comment (erlendvollset) [inline pyrightconfig.json:6]:
```suggestion
    "oryx",
    ".venv"
```
for those who like to have their .venv in the project directory, but don't want pyright to start analyzing its contents.

File: .github/workflows/python-package.yml [hunk 2]
```diff
@@ -25,6 +25,7 @@ jobs:
       - name: Install dependencies
         run: |
           python -m pip install --upgrade pip
+          npm install pyright
```
Comment (erlendvollset) [inline .github/workflows/python-package.yml:28]: Is this line necessary when we have it listed in the hook's `additional_dependencies` section?

**Review by erlendvollset — COMMENTED** [review]
very nice!

File: pyrightconfig.json (hunk 1 above)
Comment (dbrattli) [reply pyrightconfig.json:6]: Perhaps we should use include on the top-level dirs we want instead, so we only have to exclude stuff within those specific dirs.

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (dbrattli) [reply .github/workflows/python-package.yml:28]: Maybe not. I can try without.

File: pyrightconfig.json (hunk 1 above)
Comment (dbrattli) [reply pyrightconfig.json]: Fixed

File: .github/workflows/python-package.yml (hunk 2 above)
Comment (dbrattli) [reply .github/workflows/python-package.yml]: Works without the line, so have removed it.

**Review by erlendvollset — APPROVED** [review]
Lgtm

**Pull request merged**